internal/services/compute/managed_disk_resource.go

```diff
@@ -415,6 +422,14 @@ func resourceManagedDiskCreate(d *pluginsdk.ResourceData, meta interface{}) erro
 		}
 	}
 
+	if createOption == disks.DiskCreateOptionUpload {
```

Comment:
We'll want to add this to the Update function as well

Reply:
commented below that update isn't supported by SDK